pkg/apiserver/resthandler.go

```diff
@@ -470,11 +470,11 @@ func PatchResource(r rest.Patcher, scope RequestScope, typer runtime.ObjectTyper
 		ctx := scope.ContextFunc(req)
 		ctx = api.WithNamespace(ctx, namespace)
 
-		// PATCH requires same permission as UPDATE
-		if admit.Handles(admission.Update) {
+		if admit != nil && admit.Handles(admission.Patch) {
 			userInfo, _ := api.UserFrom(ctx)
 
-			err = admit.Admit(admission.NewAttributesRecord(obj, scope.Kind.GroupKind(), namespace, name, scope.Resource.GroupResource(), scope.Subresource, admission.Update, userInfo))
+			// TODO this record should include the patch itself.  Right now we can't do that because patches are not runtime.Objects
```

Comment:
> IMO that places an undue burden on admission controller authors. The patch data is almost entirely useless without the object the patch applies to, and it's silly to make the admission controller go get that and apply the patch itself when we're about to do the exact same thing.

I agree that its generally not needed, but I disagree that its never appropriate to inspect.  That's why this pull gives admission plugins the option.  You can inspect the patch if you want, but if you don't, this pull makes sure you're called with the update.